examples/xusb.c

```diff
@@ -806,6 +807,20 @@ static void print_device_cap(struct libusb_bos_dev_capability_descriptor *dev_ca
 		break;
 
 	}
+	case LIBUSB_BT_SUPERSPEED_PLUS_CAPABILITY: {
+		struct libusb_ssplus_usb_device_capability_descriptor *ssplus_usb_device_cap = NULL;
+		libusb_get_ssplus_usb_device_capability_descriptor(NULL, dev_cap, &ssplus_usb_device_cap);
+		if (ssplus_usb_device_cap) {
+			printf("    USB 3.1 capabilities:\n");
+			printf("      attributes             : %02X\n", ssplus_usb_device_cap->bmAttributes);
+			printf("      supported functionality: %04X\n", ssplus_usb_device_cap->wFunctionalitySupport);
+			for(i=0 ; i < (ssplus_usb_device_cap->bmAttributes & LIBUSB_SSPLUS_ATTRIBUTE_SSCA_MASK) + 1 ; i++) {
```

Comment:
Constraining the scope of `i` to here is advisable.

Reply:
Disagree. In every other parts of libusb, I see variables are declared at the top a la C89.

Reply:
This is historic because when Windows support was first planned it did not have C99 support in its default compiler (MSVC) leaving us at C89. It is probably safe to move to C99 as the new minimum but we can clean this up after we make such a move.

Reply:
Yes, if in doubt follow the code style of the file (or function!). We don't have an overall project style.

BTW, talking about "i" we always have spaces in assignments like "i = 0".

Reply:
>It is probably safe to move to C99...

It's more than safe, the codebase already requires C99.  In fact the Xcode project is actually configured to use C11.

>Yes, if in doubt follow the code style of the file (or function!). We don't have an overall project style.

I argue it's preferable for *new* code to do the better thing and constrain the scope of variables.

For old code, I volunteer to update it. cppcheck can identify variables that can have their scope reduced.

Reply:
Indeed. I moved the Darwin backend to C11 because all the Apple compilers have supported it for some time. As for generally, I don't see AC_PROG_CC_C99 in configure.ac but if we are willing to make the jump I support adding it and enforcing it. I have no idea how it will impact Windows though.